src/lxc/lsm/selinux.c

```diff
@@ -76,13 +74,10 @@ static int selinux_process_label_set(const char *inlabel, struct lxc_conf *conf,
 {
 	const char *label = inlabel ? inlabel : conf->lsm_se_context;
 	if (!label) {
-		if (use_default)
-			label = DEFAULT_LABEL;
-		else
+		if (!use_default)
 			return -1;
-	}
-	if (!strcmp(label, "unconfined_t"))
 		return 0;
+	}
```

Comment:
But then you leave `label` uninitialized that'll cause undefined behavior further down.

Reply:
Oh, I see.

Reply:
But you're not handling the case where the `label` that is passed in is `unconfined_t`. So the old logic still is correct.

Reply:
Ouch... true. I'll update the patch leaving the comparison with unconfined_t after the label check.